transformation/src/test/scala/org/broadinstitute/monster/hca/HcaPipelineBuilderSpec.scala

```diff
@@ -200,4 +211,83 @@ class HcaPipelineBuilderSpec extends AnyFlatSpec with Matchers {
+
+  it should "validate json schemas and throw no exceptions if files are correctly formatted" in {
+    val exampleFileContent = JsonParser.parseEncodedJson(
+      """
+        |{
+        |    "organ": {
+        |        "text": "brain",
+        |        "ontology": "astrocyte"
+        |    },
+        |    "schema_type": "biomaterial",
+        |    "biomaterial_core": {
+        |        "ncbi_taxon_id": [
+        |            9606
+        |        ],
+        |        "biomaterial_id": "Q4_DEMO-sample_SAMN02797092",
+        |        "has_input_biomaterial": "Q4_DEMO-donor_MGH30",
+        |        "biomaterial_name": "Q4_DEMO-Single cell mRNA-seq_MGH30_A01",
+        |        "supplementary_files": [
+        |            "Q4_DEMO-protocol"
+        |        ]
+        |    },
+        |    "organ_part": {
+        |        "text": "glioblastoma"
+        |    },
+        |    "genus_species": [
+        |        {
+        |            "text": "Homo sapiens",
+        |            "ontology": "NCBITaxon:9606"
+        |        }
+        |    ],
+        |    "describedBy": "https://schema.humancellatlas.org/type/biomaterial/5.1.0/specimen_from_organism"
+        |}
+        |""".stripMargin
+    )
+
+    val exampleUrlAndFile = (exampleFileContent.read[String]("describedBy"), exampleFileContent)
+
+    runWithContext(sc => HcaPipelineBuilder.validateJson(sc.parallelize(Seq(exampleUrlAndFile))))
+  }
+
+  it should "validate json schemas and throw an exception if files are incorrectly formatted" in {
+    val exampleFileContent = JsonParser.parseEncodedJson(
+      """
+        |{
+        |    "organ": {
+        |        "text": "brain",
+        |        "ontology": "astrocyte"
+        |    },
+        |    "biomaterial_core": {
+        |        "ncbi_taxon_id": [
+        |            9606
+        |        ],
+        |        "biomaterial_id": "Q4_DEMO-sample_SAMN02797092",
+        |        "has_input_biomaterial": "Q4_DEMO-donor_MGH30",
+        |        "biomaterial_name": "Q4_DEMO-Single cell mRNA-seq_MGH30_A01",
+        |        "supplementary_files": [
+        |            "Q4_DEMO-protocol"
+        |        ]
+        |    },
+        |    "organ_part": {
+        |        "text": "glioblastoma"
+        |    },
+        |    "genus_species": [
+        |        {
+        |            "text": "Homo sapiens",
+        |            "ontology": "NCBITaxon:9606"
+        |        }
+        |    ],
+        |    "describedBy": "https://schema.humancellatlas.org/type/biomaterial/5.1.0/specimen_from_organism"
+        |}
+        |""".stripMargin
+    )
+
+    val exampleUrlAndFile = (exampleFileContent.read[String]("describedBy"), exampleFileContent)
+
+    the[Exception] thrownBy runWithContext { sc =>
+      HcaPipelineBuilder.validateJson(sc.parallelize(Seq(exampleUrlAndFile)))
+    } should have message "java.lang.Exception: Data does not conform to schema: NonEmptyList(#: required key [schema_type] not found)"
```

Comment:
You could shorten this to be more flexible, in case there's any weird heuristics in the parser that might change the datatype used:
```suggestion
    } should include "Data does not conform to schema"
```

Reply:
Oh, just noticed this is some magic helper for exceptions. Not sure if `include` works there.

Reply:
Yeah I was following an example from the scala testing docs